tooling/ssa_fuzzer/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+#![allow(dead_code)]
+pub mod builder;
+pub mod compiler;
+pub mod config;
+pub mod helpers;
+pub mod runner;
+
+#[cfg(test)]
+mod tests {
+    use crate::builder::FuzzerBuilder;
+    use crate::config;
+    use crate::runner::run_and_compare;
+    use acvm::FieldElement;
+    use acvm::acir::native_types::{Witness, WitnessMap};
+    use noirc_evaluator::ssa::ir::{map::Id, types::Type, value::Value};
+    use rand::RngCore;
+
+    struct TestHelper {
+        acir_builder: FuzzerBuilder,
+        brillig_builder: FuzzerBuilder,
+    }
+
+    impl TestHelper {
+        fn new(typ: Type) -> Self {
+            let mut acir_builder = FuzzerBuilder::new_acir();
+            let mut brillig_builder = FuzzerBuilder::new_brillig();
+            acir_builder.insert_variables(typ.clone());
+            brillig_builder.insert_variables(typ.clone());
+            Self { acir_builder, brillig_builder }
+        }
+
+        fn insert_instruction_double_arg(
+            &mut self,
+            instruction: fn(&mut FuzzerBuilder, Id<Value>, Id<Value>) -> Id<Value>,
```

Comment:
Can you add some docs about what the `instruction` does? `acir_result` sounds like it could be execution result, but I guess it's the ID of the returned witness. Maybe call it `acir_return` instead of `_result`, so it doesn't give `Result<_, _>` vibes.

You could create a type alias for `fn(&mut FuzzerBuilder, Id<Value>, Id<Value>) -> Id<Value>` to capture the explanation in one place.